File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int key = target - nums[i];
            if (map.containsKey(key)) {
                result[1] = i;
                result[0] = map.get(key);
                return result;
            } else {
                map.put(nums[i], i);
            }
        }
        return result;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        map<int, int> dict;

        for (int i = 0; i < nums.size(); i++) {
            int key = target - nums[i];
            if (dict.count(key) > 0) {
                result.push_back(dict[key]);
                result.push_back(i);
                break;
            } else {
                dict[nums[i]] = i;
            }
        }
        return result;
    }
};